### ISSUE TITLE
dhall.org imports demo broken

### ISSUE THREAD

**Issue by ccressent** [dhall-lang/dhall-lang#579]
I was giving the demo a try on https://dhall-lang.org, and I think the last tab with the imports example is broken. More specifically, I don't think the output pane on the right is properly refreshed.

On a fresh page load, clicking on the "Imports" tab shows this:

```
...
in  {- Try generating 20 users instead of 10 -}
    generate 10 Config buildUser
```

I'm assuming this should generate 10 users in the output pane on the right, but it only shows whatever output was already in the output pane. On a fresh page load, that's the first example ("Hello, world!"): 

```
home: /home/bill
privateKey: /home/bill/id_ed25519
publicKey: /home/blil/id_ed25519.pub
```

Similarly, navigating to another tab, say "Functions", and then back to "Imports" leaves the output of "Functions" in the right pane.

Back to the "Imports" tab, changing the integer value, as suggested by the comment, doesn't really work either: `generate 0 ...` and `generate 1 ...` give the expected output, but most other integers give the following error:

```
generate : ∀(n : Natural) → ∀(a : Type) → ∀(f : Natural → a) → List a
makeUser : ∀(user : Text) → { home : Text, privateKey : Text, publicKey : Text }
buildUser : ∀(index : Natural) → { home : Text, privateKey : Text, publicKey : Text }

Error: Wrong type of function argument

- Natural
+ Type

39:     generate  Config 

(input):39:5
```

The integer argument is missing, and it looks like the changes happening in the left pane are not always properly taken into account.

I've tried in both Chrome and Firefox, same result.

**Comment by Gabriel439** [comment]
Yeah, I can reproduce this myself and I have a pretty good guess to why this is happening.

I ran into exactly this issue when first implementing the live demo and it was due to a bug in GHCJS's handling of Haskell's `Natural` type, which I fixed with this change:

https://github.com/dhall-lang/dhall-haskell/commit/cf69f5a953a3264bd4035ed9e60d466ed1267a50#diff-93a41a07dc9e35c0a2445532e379eedfR1612

Then, recently `dhall` switched to a much faster evaluation implementation, but in the process that GHCJS work-around was lost, which is why the issue resurfaced.  I'll take a stab at fixing this tonight.
